dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java

```diff
@@ -278,33 +279,38 @@ public Item createTemplateItem(Context context, Collection collection) throws SQ
 
     @Override
     public Iterator<Item> findAll(Context context) throws SQLException {
-        return itemDAO.findAll(context, true);
+        return itemDAO.findAll(context.getSession(), true);
+    }
+
+    @Override
+    public Iterator<Item> findAll(Session session) throws SQLException {
```

Comment:
@mwoodiupui : One thought for next week.  While I like the overall approach of this PR, I find it odd that *some `Service` methods* take in `Context` while others take in `Session`.  For instance here, I see we have a new `findAll(Session)`, but the `find(Context, UUID)` only supports Context.

I suspect this `findAll(Session)` is just to illustrate an example, but I wanted to call this out. It's confusing to see some `find*` methods only supporting `Context`, while others support using either `Context` or `Session`.

The overarching question is that I'm not yet sure how to make the decision of which methods should also support `Session` (is it *all*? Is it some?), and which should only allow developers to use `Context`.  I'm not expecting any immediate answer.  Just wanting to express a question about the scope of this work going forward (i.e. where do we *stop* replacing Context with Session in our API layer?)

Reply:
You raise an important question.  Another way to ask it is:  where do we have layer boundaries, and what should each layer "know"?

My approach so far is that DAOs are part of the storage layer, and storage should not know anything about Context.  I added a necessary method to one DSO type as an example, but have not considered whether DSO Services are a layer unto themselves.

It is possible that, at least, all of the Service `find*` methods could use Session rather than Context, but I have not examined that thoroughly.

Reply:
I am finding that there are a handful of `find*` methods that really do need the `Context` but the vast majority do not.